src/Microsoft.ML.Core/Utilities/VBufferUtils.cs

```diff
@@ -908,7 +908,7 @@ public static void CreateMaybeSparseCopy<T>(ref VBuffer<T> src, ref VBuffer<T> d
         /// where necessary depending on whether this is an inner or outer join of the
         /// indices of <paramref name="src"/> on <paramref name="dst"/>.
         /// </summary>
-        private static void ApplyWithCoreCopy<TSrc, TDst>(ref VBuffer<TSrc> src, ref VBuffer<TDst> dst, ref VBuffer<TDst> res, PairManipulatorCopy<TSrc, TDst> manip, bool outer)
+        private static void ApplyWithCoreCopy<TSrc, TDst>(in VBuffer<TSrc> src, ref VBuffer<TDst> dst, ref VBuffer<TDst> res, PairManipulatorCopy<TSrc, TDst> manip, bool outer)
```

Comment:
>ref [](start = 80, length = 3)

can it be "in"?

Reply:
This `dst` (and the two calling methods above) cannot be made `in` because of the case at this line of code:

[https://github.com/dotnet/machinelearning/blob/ef2bade365a90d926a944ebfc424b8a84374921c/src/Microsoft.ML.Core/Utilities/VBufferUtils.cs#L1089-L1093](https://github.com/dotnet/machinelearning/blob/ef2bade365a90d926a944ebfc424b8a84374921c/src/Microsoft.ML.Core/Utilities/VBufferUtils.cs#L1089-L1093)

This method actually modifies `dst` in that case, by making it dense. Thus, the `dst` can't be marked `in`.

---
In reply to: [229498975](https://github.com/dotnet/machinelearning/pull/1454#discussion_r229498975) [](ancestors = 229498975)